document/service.py

```diff
@@ -0,0 +1,91 @@
+from abc import ABC, abstractmethod
+import io
+
+from loguru import logger
+import boto3
+from botocore.exceptions import BotoCoreError, NoCredentialsError, PartialCredentialsError
+
+from document.document import ObjectNameError
+from document.dto import DocumentCreate, DocumentFilter, DocumentResponse, DocumentUpdate
+from document.repository import DocumentRepository
+
+
+class DocumentService(ABC):
+    @abstractmethod
+    def upload_document(self, file, object_name):
+        pass
+
+    @abstractmethod
+    def get_documents(self, filter: DocumentFilter) -> list[DocumentResponse]:
+        pass
+      
+    @abstractmethod
+    def get_document_by_name(self, object_name: str) -> DocumentResponse:
+        pass
+      
+    @abstractmethod
+    def get_document_by_id(self, doc_id: str) -> DocumentResponse:
+        pass
+
+    @abstractmethod
+    def create_document(self, request: DocumentCreate):
+        pass
+
+    # note: Not implemented to ensure vector database and document entries are not out of sync
+    # @abstractmethod
+    # def update_document(self, doc_id: str, request: DocumentUpdate):
+    #     pass
+    
+    # @abstractmethod
+    # def delete_document(self, doc_id: str):
+    #     pass
+      
+class DocumentServiceV1(DocumentService):
+    def __init__(self, aws_access_key_id, aws_secret_access_key, bucket_name, aws_region_name, repository: DocumentRepository):
+        super().__init__()
+        self.repository = repository
+        self.s3_client = boto3.client(
+            's3',
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=aws_region_name,
+            endpoint_url="https://broom-magang.s3.ap-southeast-3.amazonaws.com"
```

Comment:
I know I showed an example prior, but do you think this might be worth putting to a `.env` config?

Reply:
Yes thanks for the suggestion